include/arch/arm/cortex_m/scripts/linker.ld

```diff
@@ -339,15 +334,41 @@ SECTIONS
 
 	__app_last_address_used = .;
 
-	/* Pad out application ram area to make MPU friendly */
-	SECTION_PROLOGUE(app_pad, (NOLOAD OPTIONAL),)
-	{
-		. = ALIGN(_app_data_align);
-	} GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
+	/* Align the end of the application memory
+	 * with MPU alignment requirement.
+	 */
+#ifdef CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT
+	. = ALIGN( 1 << LOG2CEIL(__app_ram_size));
+#endif
+	. = ALIGN(_region_min_align);
 	__app_ram_end = .;
 	__app_ram_size = __app_ram_end - __app_ram_start;
 #endif /* CONFIG_APPLICATION_MEMORY */
 
+#if defined(CONFIG_APP_SHARED_MEM)
+#if defined(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT)
```

Comment:
I don't get why we still need to get two different strategies for application shared memory depending if the MPU requires power of two alignement or not.

Reply:
@AdithyaBaglody could you advice here?
We could possibly try to use _only_ the script for both cases of MPU architectures.

(The difference with APPLICATION_MEMORY is that, in this case, we have multiple sections.)